internal/api/v1beta1/user.go

```diff
@@ -120,39 +117,53 @@ func (h Handler) CreateUser(ctx context.Context, request *shieldv1beta1.CreateUs
 		return nil, grpcBadBodyError
 	}
 
-	for k := range metaDataMap {
-		urn := servicedata.CreateURN(h.serviceDataConfig.DefaultServiceDataProject, k)
-		key, err := h.serviceDataService.GetKeyByURN(ctx, urn)
-		if err != nil {
-			return nil, err
-		}
-
-		permission, err := h.relationService.CheckPermission(ctx, currentUser, namespace.Namespace{ID: schema.ServiceDataKeyNamespace}, key.ResourceID, action.Action{ID: schema.EditPermission})
-		if err != nil {
-			return nil, err
-		}
-		if !permission {
-			return nil, status.Error(codes.PermissionDenied, fmt.Sprintf("you are not authorized to update %s key", k))
-		}
-	}
-
-	// TODO might need to check the valid email form
-	newUser, err := h.userService.Create(ctx, user.User{
-		Name:     request.GetBody().GetName(),
-		Email:    email,
-		Metadata: nil,
-	})
+	currentUser, err := h.userService.FetchCurrentUser(ctx)
 	if err != nil {
 		logger.Error(err.Error())
 		switch {
-		case errors.Is(err, user.ErrConflict):
-			return nil, grpcConflictError
-		case errors.Is(errors.Unwrap(err), user.ErrKeyDoesNotExists):
-			return nil, grpcBadBodyError
-		case errors.Is(err, user.ErrInvalidEmail),
-			errors.Is(err, user.ErrMissingEmail):
-			return nil, grpcUnauthenticated
+		case errors.Is(err, user.ErrInvalidEmail):
+			callerEmail, ok := user.GetEmailFromContext(ctx)
+			if !ok {
+				return nil, grpcUnauthenticated
+			}
+			if callerEmail != email {
+				return nil, grpcUnauthenticated
+			}
+			userSelfOnboard = true
+			if !h.checkIsPublicEditable(ctx, metaDataMap) {
```

Comment:
Instead of calling it public, what about calling it wildcard?

checkIsWildcardEditable